pennylane/transforms/qcut.py

```diff
@@ -18,10 +18,73 @@
 
 from networkx import MultiDiGraph
 from pennylane.measure import MeasurementProcess
-from pennylane.operation import Operator, Tensor
+from pennylane.operation import Observable, Operation, Operator, Tensor
+from pennylane.ops.qubit.non_parametric_ops import WireCut
 from pennylane.tape import QuantumTape
 
 
+class MeasureNode(Operation):
+    """Placeholder node for measurement operations"""
+
+    num_wires = 1
+    grad_method = None
+
+
+class PrepareNode(Operation):
+    """Placeholder node for state preparations"""
+
+    num_wires = 1
+    grad_method = None
+
+
+def replace_wire_cut_node(node: WireCut, graph: MultiDiGraph):
+    """
+    Replace a `WireCut` nodes in the graph with `MeasureNode` and `PrepareNode`
+    """
+    predecessors = graph.pred[node]
+    successors = graph.succ[node]
+
+    predecessor_on_wire = {}
+    for op, data in predecessors.items():
+        for d in data.values():
+            wire = d["wire"]
+            predecessor_on_wire[wire] = op
+
+    successor_on_wire = {}
+    for op, data in successors.items():
+        for d in data.values():
+            wire = d["wire"]
+            successor_on_wire[wire] = op
+
+    order = graph.nodes[node]["order"]
+    graph.remove_node(node)
+
+    for wire in node.wires:
+        predecessor = predecessor_on_wire.get(wire, None)
+        successor = successor_on_wire.get(wire, None)
+
+        meas = MeasureNode(wires=wire)
+        prep = PrepareNode(wires=wire)
+        graph.add_node(meas, order=order)
+        graph.add_node(prep, order=order)
```

Comment:
Maybe we should document with a comment the decision here to introduce a degeneracy in the order, i.e., now we'll have two nodes with the same order but whose order can be inferred because meas always precedes prep.